C/mcal/SPI/SPI_config.h

```diff
@@ -0,0 +1,47 @@
+#ifndef SPI_CONF_H
+#define SPI_CONF_H
+/**
+ * options:         1-MSB_FIRST
+ *                  2-LSB_FIRST
+*/
+#define SPI_DATA_ORDER         (MSB_FIRST)
```

Comment:
It will be good to handle error, as you may know macros are very hard to debug so if we handle invalid cases that will reduce the bugs caused by choosing invalid values. e.g.
```
#if !(SPI_DATA_ORDER != SPI_MSB_FIRST)
#if !(SPI_DATA_ORDER != SPI_LSB_FIRST)
#error "[SPI_CONFIG]: Invalid value for  SPI_DATA_ORDER "
#endif 
#endif